### PR TITLE
enzyme: drop llvm 10 and 11

### PR DESCRIPTION
**Pull request by tgymnich**
drop now unsupported llvm versions

### CODE REVIEW

File: var/spack/repos/builtin/packages/enzyme/package.py [hunk 1]
```diff
@@ -34,7 +35,8 @@ class Enzyme(CMakePackage):
     depends_on("llvm@7:12", when="@0.0.13:0.0.15")
     depends_on("llvm@7:14", when="@0.0.32:0.0.47")
     depends_on("llvm@7:14", when="@0.0.48:0.0.68")
-    depends_on("llvm@9:16", when="@0.0.69:")
+    depends_on("llvm@9:16", when="@0.0.69:0.0.79")
+    depends_on("llvm@11:16", when="@0.0.80:")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/enzyme/package.py:39]: FWIW. You could reduce the need to make changes with updates to `llvm` by removing `16` from the version range.

File: var/spack/repos/builtin/packages/enzyme/package.py [hunk 2]
```diff
@@ -34,7 +35,8 @@ class Enzyme(CMakePackage):
     depends_on("llvm@7:12", when="@0.0.13:0.0.15")
     depends_on("llvm@7:14", when="@0.0.32:0.0.47")
     depends_on("llvm@7:14", when="@0.0.48:0.0.68")
-    depends_on("llvm@9:16", when="@0.0.69:")
+    depends_on("llvm@9:16", when="@0.0.69:0.0.79")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/enzyme/package.py:38]: Did you intend to drop unsupported 10 and 11 from here as well?  Or does that just apply for versions `0.0.80` on?

**Review by tldahlgren — APPROVED** [review]
Confirmed the version sha256 for the new version, rest LGTM though I have a couple of questions.

File: var/spack/repos/builtin/packages/enzyme/package.py (hunk 2 above)
Comment (tgymnich) [reply var/spack/repos/builtin/packages/enzyme/package.py:38]: 10 and 11 were dropped from 0.0.80 onwards.

File: var/spack/repos/builtin/packages/enzyme/package.py (hunk 1 above)
Comment (tgymnich) [reply var/spack/repos/builtin/packages/enzyme/package.py:39]: llvm 17 is actually not yet supported

**Pull request merged**